simple-game-server-go/internal/game/allocated.go

```diff
@@ -120,3 +126,38 @@ func (g *Game) handleClient(client *net.TCPConn) {
 		}
 	}
 }
+
+// readyForPlayers waits 20s then reports that the game is ready for players.
+//
+// This is to simulate a game server waiting for any initialization to complete
+// before reporting that it is ready to accept players.
+//
+// You can optionally configure the timeout by setting the "readyTimeout"
+// configuration variable in the build configuration to a duration string
+// recognised by `time.ParseDuration`, e.g. "30s".
+func (g *Game) readyForPlayers() {
+	g.logger.Info("ready for players")
+
+	timeout := defaultReadyTimeout
+
+	if g.Config().Extra["readyTimeout"] != "" {
+		t, err := time.ParseDuration(g.Config().Extra["readyTimeout"])
+		if err != nil {
+			g.logger.WithError(err).Error("parsing ready timeout")
+		} else {
+			timeout = t
+		}
+	}
+
+	select {
+	case <-time.After(timeout):
+		g.logger.Info("ready timeout elapsed, reporting ready")
+	case <-g.OnDeallocate():
+		g.logger.Info("server was deallocated, don't report ready")
+		return
+	}
+
+	if err := g.ReadyForPlayers(context.Background()); err != nil {
```

Comment:
Sorry, I made a bit of an error here in my suggestion, we should use a timeout context, otherwise this will hang.

```suggestion
    ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
    defer cancel()
	if err := g.ReadyForPlayers(ctx); err != nil {
```